scripts/docker/update_dockers_json.py

```diff
@@ -0,0 +1,212 @@
+"""
+The intent of this script is to update `dockers.json` with
+the images rebuilt and pushed by `build_docker.py`. Since for
+a given list of images, `build_docker.py` may rebuild additional
+images, the list of all the built and pushed images may be different
+from the user-requested list. However, the `build_docker.py` does
+not export the list of all the built images, hence, this script
+takes a bruteforce approach and checks for all the images if they
+are updated by a given tag (i.e., the tag used by the `build_docker.py`
+to tag all the updated images).
+
+This script is mainly developed to be used in Github actions
+for the CI/CD pipeline, and is a temporary solution until
+`build_docker.py` is updated to export the list of all the built and
+pushed images.
+"""
+
+import argparse
+import copy
+import json
+import os
+from enum import Enum
+from pathlib import Path
+from subprocess import Popen, PIPE
+
+
+COLOR_RED = "\033[91m"
+COLOR_GREEN = "\033[92m"
+COLOR_ENDC = "\033[0m"
+
+
+class Status(Enum):
+    NO_MARK = 0
+    NEUTRAL = 1
+    SUCCESS = 2
+    FAILED = 3
+
+
+def pprint_log(msg: str, status=Status.NO_MARK, line_break=True):
+    if status != Status.NO_MARK:
+        # "\N{<unicode name of a character>}"
+        # See https://unicode.org for other characters.
+        if status == Status.NEUTRAL:
+            mark = "\N{black star}"
+        elif status == Status.SUCCESS:
+            mark = "\N{heavy check mark}"
+        else:
+            mark = "\N{heavy ballot x}"
+        msg = f"{mark}\t{msg}"
+
+    if status in [Status.SUCCESS, Status.FAILED]:
+        txt_color = COLOR_GREEN if status == status.SUCCESS else COLOR_RED
+        msg = f"{txt_color}{msg}{COLOR_ENDC}"
+    print(msg, end="\n" if line_break else "", flush=True)
+
+
+def check_image_exist(image: str) -> (bool, str):
```

Comment:
The type hints here aren't quite right, since the second output ("error") is sometimes None, which is not a string.
At the top include `from typing import Optional`
```suggestion
def check_image_exist(image: str) -> (bool, Optional[str]):
```
Alternatively you could just always return the decoded stderr string.